crates/icrate/examples/metal.rs

```diff
@@ -0,0 +1,344 @@
+
+            let mut scene_properties_data = SceneProperties {
+                time: unsafe { self.start_date.timeIntervalSinceNow() } as f32,
+            };
+            let scene_properties_bytes =
+                unsafe { NonNull::new_unchecked(&mut scene_properties_data) }
+                    .cast::<core::ffi::c_void>();
+            unsafe {
+                encoder.setVertexBytes_length_atIndex(
+                    scene_properties_bytes,
+                    core::mem::size_of::<SceneProperties>(),
+                    0,
+                )
+            };
+
+            let vertex_input_data: &mut [VertexInput] = &mut [
+                VertexInput {
+                    position: Position {
+                        x: -f32::sqrt(3.0) / 4.0,
+                        y: -0.25,
+                        z: 0.,
+                    },
+                    color: Color {
+                        r: 1.,
+                        g: 0.,
+                        b: 0.,
+                    },
+                },
+                VertexInput {
+                    position: Position {
+                        x: f32::sqrt(3.0) / 4.0,
+                        y: -0.25,
+                        z: 0.,
+                    },
+                    color: Color {
+                        r: 0.,
+                        g: 1.,
+                        b: 0.,
+                    },
+                },
+                VertexInput {
+                    position: Position {
+                        x: 0.,
+                        y: 0.5,
+                        z: 0.,
+                    },
+                    color: Color {
+                        r: 0.,
+                        g: 0.,
+                        b: 1.,
+                    },
+                },
+            ];
+            let vertex_input_bytes =
+                unsafe { NonNull::new_unchecked(vertex_input_data) }.cast::<core::ffi::c_void>();
+            unsafe {
+                encoder.setVertexBytes_length_atIndex(
+                    vertex_input_bytes,
+                    vertex_input_data.len() * core::mem::size_of::<VertexInput>(),
+                    1,
+                )
+            };
+
+            encoder.setRenderPipelineState(&self.pipeline_state);
+            unsafe {
+                encoder.drawPrimitives_vertexStart_vertexCount(MTLPrimitiveTypeTriangle, 0, 3)
+            };
+            encoder.endEncoding();
+            command_buffer.presentDrawable(&current_drawable);
+            command_buffer.commit();
+        }
+
+        #[method(mtkView:drawableSizeWillChange:)]
+        #[allow(non_snake_case)]
+        unsafe fn mtkView_drawableSizeWillChange(&self, _view: &MTKView, _size: NSSize) {
+            // println!("mtkView_drawableSizeWillChange");
+        }
+    }
+);
+
+unsafe impl NSObjectProtocol for Delegate {}
+
+#[cfg(target_os = "macos")]
```

Comment:
I think the whole example might be macOS only?

Though maybe we could make it work on iOS as well, by implementing `UIApplicationDelegate` too, and swapping out the `window` type?